sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs

```diff
@@ -109,12 +125,14 @@ public class ProcessMessageEventArgs : EventArgs
             _receiver = manager?.Receiver;
             CancellationToken = cancellationToken;
 
-            messageLockLostCancellationSource = new CancellationTokenSource();
-            MessageLockCancellationToken = messageLockLostCancellationSource.Token;
-            messageLockLostCancellationSource.CancelAfterLockExpired(Message);
+            MessageLockLostCancellationSource = new CancellationTokenSource();
+            MessageLockCancellationToken = MessageLockLostCancellationSource.Token;
+
+            MessageLockCancellationToken.Register(() => OnMessageLockLostAsync(new MessageLockLostEventArgs(message, LockLostException)));
```

Comment:
Who is responsible for disposing this token registration?

Reply:
Good catch - this should have been deleted as it was replaced by RegisterMessageLockLostHandler